### DISCUSSION TITLE
Concourse api

### DISCUSSION THREAD

**Discussion by richardforjoejnr** [concourse/concourse#8622] [Help & Support]
Hi, I wanted to find out if concourse has an api available to perform certain tasks like the fly command?

So being able to send a post command to a concourse url to trigger a pipeline and pass it some variables via an api endpoint? No via fly cli?

**Comment by richardforjoejnr** [comment]
Also do you have any resource/ articles on integrating slack and concourse? Would be great to be able to trigger a pipeline via slack.

**Reply by marco-m** [reply]
Concourse has an HTTP API, although there is no commitment to backward compatibility. See https://github.com/concourse/concourse/blob/master/atc/routes.go

> Would be great to be able to trigger a pipeline via slack

It is impossible to trigger a pipeline, what you can do is to trigger a job. If the pipeline has a certain structure, then triggering a specific job can actually give the result of "triggering a pipeline".

See `fly trigger-job` (from which you could find the equivalent HTTP API endpoint).
https://concourse-ci.org/jobs.html#fly-trigger-job

**Comment by gsaslis** [comment]
Just to add a +1 here to the need for a better-documented HTTP API for a clearer integration between concourse and other systems. The `fly` Command Line Interface (CLI) is great for desktop use, but when talking about integrations with other backend systems, the API certainly seems like the better tool for the job, no ?  :)
